### PR TITLE
Mvvsubbu databricks microsoft.databricks 2024 02 01 preview

### PR DESCRIPTION
**Pull request by mvvsubbu**
# ARM (Control Plane) API Specification Update Pull Request 

> [!TIP]
> Overwhelmed by all this guidance? See the `Getting help` section at the bottom of this PR description.

> [!NOTE]
> As of January 2024 there is no PR assignee. This is expected. See https://aka.ms/azsdk/pr-arm-review.

## PR review workflow diagram

Please understand this diagram before proceeding. It explains how to get your PR approved & merged.

diagram

<details>
<summary> Click here to see the details of Step 1 </summary>

### Breaking changes review (Diagram Step 1)

If the automation determines you have breaking changes, i.e. Step 1 from the diagram applies to you,
  you must follow the [breaking changes process].  
  **IMPORTANT** This applies even if:
- The tool fails while it shouldn't, e.g. due to runtime exception, or incorrect detection of breaking changes.
- You believe there is no need for you to request breaking change approval, for any reason. 
    Such claims must be reviewed, and the process is the same.

</details>

<details> 
<summary> Click here to see the details of Step 2 </summary>

### ARM API changes review (Diagram Step 2)

- For help with Step 2 / "ARM review", see https://aka.ms/azsdk/pr-arm-review.

</details>

<details>
<summary> Click here to see the diagram footnotes</summary>

### Diagram footnotes

[1] See [ARM review queue] (for **PR merge** queues, see [2]).  
[2] [public repo merge queue], [private repo merge queue] (for **ARM review** queue, [1])  
The ARM reviewer on-call engineer visits the merge queue twice a day, so the approximate ETA for merges is 12 - 24 hours.

</details>

## Purpose of this PR

What's the purpose of this PR? Check the specific option that applies. This is **mandatory**!

  - [ ] New resource provider.
  - [x] New API version for an existing resource provider. (If API spec is not defined in TypeSpec, the PR should have been generated using [OpenAPI Hub](https://aka.ms/openapihub)).
  - [ ] Update existing version for a new feature. (This is applicable only when you are revising a private preview API version.)
  - [ ] Update existing version to fix OpenAPI spec quality issues in S360.
  - [ ] Other, please clarify:
    - _edit this with your clarification_

## Due diligence checklist

To merge this PR, you **must** go through the following checklist and confirm you understood 
and followed the instructions by checking all the boxes:

- [x] I confirm this PR is modifying Azure Resource Manager (ARM) related specifications, and not data plane related specifications.
- [x] I have reviewed following [Resource Provider guidelines](https://aka.ms/rpguidelines), including
  [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and
  [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md) (estimated time: 4 hours).  
  I understand this is required before I can proceed to the diagram Step 2, "ARM API changes review", for this PR.

## Additional information

<details>
<summary> Viewing API changes</summary>

For convenient view of the API changes made by this PR, refer to the URLs provided in the table 
in the `Generated ApiView` comment added to this PR. You can use ApiView to show API versions diff. 

</details>
<details>
<summary>Suppressing failures</summary>

If one or multiple validation error/warning suppression(s) is detected in your PR, please follow the 
[suppressions guide](https://aka.ms/azsdk/pr-suppressions) to get approval.

</details>

## Getting help

- First, please carefully read through this PR description, from top to bottom. Please fill out the `Purpose of this PR` and `Due diligence checklist`.
- To understand what you must do next to merge this PR, see the `Next Steps to Merge` comment. It will appear within few minutes of submitting this PR and will continue to be up-to-date with current PR state.
- For guidance on fixing this PR CI check failures, see the hyperlinks provided in given failure 
  and https://aka.ms/ci-fix.
- For help with PR workflow diagram Step 2 (ARM review), see https://aka.ms/azsdk/pr-arm-review.
- If the PR CI checks appear to be stuck in `queued` state, please add a comment with contents `/azp run`.
  This should result in a new comment denoting a `PR validation pipeline` has started and the checks should be updated after few minutes.
- If the help provided by the previous points is not enough, post to https://aka.ms/azsdk/support/specreview-channel and link to this PR.

[ARM review queue]: https://aka.ms/azsdk/pr-arm-review#about-the-arm-review-queue
[public repo merge queue]: https://github.com/Azure/azure-rest-api-specs/pulls?q=is%3Aopen+is%3Apr+label%3AMergeRequested+draft%3Afalse+sort%3Acreated-asc
[private repo merge queue]: https://github.com/Azure/azure-rest-api-specs-pr/pulls?q=is%3Aopen+is%3Apr+label%3AMergeRequested+-label%3AApproved-OkToMerge+draft%3Afalse+sort%3Acreated-asc
[breaking changes process]: https://eng.ms/docs/cloud-ai-platform/azure-core/azure-core-pm-and-design/trusted-platform-pm-karimb/service-lifecycle-and-actions-team/service-lifecycle-actions-team/apex/media/launchingproductbreakingchanges#breaking-change-process-1

### CODE REVIEW

**Comment by mvvsubbu** [comment]
Thanks @maririos for helping on the base commit missing issue and filing a bug for the same.

File: specification/databricks/resource-manager/Microsoft.Databricks/preview/2024-02-01-preview/databricks.json [hunk 1]
```diff
@@ -0,0 +1,1981 @@
...
+        "isUcEnabled": {
+          "readOnly": true,
+          "description": "Indicates whether unity catalog enabled for the workspace or not.",
+          "type": "boolean"
+        },
+        "accessConnector": {
+          "description": "Access Connector Resource that is going to be associated with Databricks Workspace",
+          "type": "object",
+          "properties": {
+            "id": {
```
Comment (visingla-ms) [inline specification/databricks/resource-manager/Microsoft.Databricks/preview/2024-02-01-preview/databricks.json:881]: The name "id" of a property should not be used in nested definitions. Please use "connectorId" or "resourceId" or similar here. Also, use [https://github.com/Azure/autorest/blob/main/docs/extensions/readme.md#x-ms-arm-id-details](https://github.com/Azure/autorest/blob/main/docs/extensions/readme.md#x-ms-arm-id-details) here.

File: specification/databricks/resource-manager/Microsoft.Databricks/preview/2024-02-01-preview/databricks.json [hunk 2]
```diff
@@ -0,0 +1,1981 @@
...
+                "SystemAssigned",
+                "UserAssigned"
+              ],
+              "x-ms-enum": {
+                "modelAsString": true,
+                "name": "IdentityType"
+              }
+            },
+            "userAssignedIdentityId": {
+              "type": "string",
```
Comment (visingla-ms) [inline specification/databricks/resource-manager/Microsoft.Databricks/preview/2024-02-01-preview/databricks.json:898]: Please use [https://github.com/Azure/autorest/blob/main/docs/extensions/readme.md#x-ms-arm-id-details](https://github.com/Azure/autorest/blob/main/docs/extensions/readme.md#x-ms-arm-id-details) here.

File: specification/databricks/resource-manager/Microsoft.Databricks/preview/2024-02-01-preview/databricks.json (hunk 1 above)
Comment (mvvsubbu) [reply specification/databricks/resource-manager/Microsoft.Databricks/preview/2024-02-01-preview/databricks.json:881]: Thanks for sharing this recommendation, however this feature has gone through private preview review and got approved for using id and now are publishing to public preview. Please refer the below PR ARM approved it, going forward we will keep this guidance in mind and use appropriate naming. 


https://github.com/Azure/azure-rest-api-specs-pr/pull/11313

**Comment by mvvsubbu** [comment]
@visingla-ms can you check the review

File: specification/databricks/resource-manager/Microsoft.Databricks/preview/2024-02-01-preview/databricks.json (hunk 1 above)
Comment (mentat9) [reply specification/databricks/resource-manager/Microsoft.Databricks/preview/2024-02-01-preview/databricks.json:881]: I agree it's unfortunate this wasn't caught sooner. Please note for future: top-level ARM resource envelope property names (name, id, type, etc.) should never be reused within the property bag. I'm leaving it to you whether to make this change right now or not: the recommendation from ARM is to make it now before your API goes public.

**Comment by mentat9** [comment]
@mvvsubbu - Signing off for ARM with comments. Please take a look and consider making the change.

File: specification/databricks/resource-manager/Microsoft.Databricks/preview/2024-02-01-preview/databricks.json (hunk 1 above)
Comment (mvvsubbu) [reply specification/databricks/resource-manager/Microsoft.Databricks/preview/2024-02-01-preview/databricks.json:881]: While I wish to consider making the changes, but that will cause the existing customers breaking change. So I will leave this as is. Going forward, I will make sure our team is educated on this and follow the guidance.

**Review by mentat9 — APPROVED** [review]

**Pull request merged**